x/metadata/keeper/msg_server.go

```diff
@@ -60,61 +60,58 @@ func (k msgServer) ChangeOwnership(
 	return nil, fmt.Errorf("not implemented")
 }
 
-func (k msgServer) AddScopeSpecification(
+func (k msgServer) AddScope(
 	goCtx context.Context,
-	msg *types.MsgAddScopeSpecificationRequest,
-) (*types.MsgAddScopeSpecificationResponse, error) {
+	msg *types.MsgAddScopeRequest,
+) (*types.MsgAddScopeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	existing, _ := k.GetScopeSpecification(ctx, msg.Specification.SpecificationId)
-	if err := k.ValidateScopeSpecUpdate(ctx, existing, *msg.Specification, msg.Signers); err != nil {
+	existing, _ := k.GetScope(ctx, msg.Scope.ScopeId)
+	if err := k.ValidateScopeUpdate(ctx, existing, msg.Scope, msg.Signers); err != nil {
 		return nil, err
 	}
 
-	k.SetScopeSpecification(ctx, *msg.Specification)
+	k.SetScope(ctx, msg.Scope)
 
 	ctx.EventManager().EmitEvent(
 		sdk.NewEvent(
-			sdk.EventTypeMessage,
```

Comment:
the `message` events are separate from the keeper events ... i.e. there is an event when a msg comes in, there is also an event when a scope is added... the two are separate.  There is a ticket to handle this work already #88 so there isn't a need to make changes here to resolve this issue.

Reply:
I reordered the funcs in this file to match their ordering in the `tx.proto` `service Msg` section. So this chunk is kind of a mess as a diff.